storage_service/locations/signals.py

```diff
@@ -44,4 +48,23 @@ def report_failed_fixity_check(sender, **kwargs):
 Full failure report (in JSON format):
 {}
 """.format(kwargs["uuid"], kwargs["location"], kwargs["report"])
+    package = models.Package.objects.get(uuid=kwargs['uuid'])
+
+    log = models.FixityLog()
+    log.package = package
+    report_data = json.loads(kwargs['report'])
+    log.error_details = report_data['message']
+    log.save()
```

Comment:
You can also create an object in one line/atomically with [`FixityLog.objects.create()`](https://docs.djangoproject.com/en/dev/ref/models/querysets/#create)